### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by MirSzabo**

### CODE REVIEW

File: nodejs/week4/spam-detector/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+}
+
+const spamWords = ["viagra", "offer", "free", "business proposal"];
+
+class SpamDetector {
+  isSpam(email) {
+    const subject = email.subject;
+    const body = email.body;
+    const string = subject + body;
```
Comment (Hatko) [inline nodejs/week4/spam-detector/index.js]: `string` doesn't represent the content of the `const` really well. Maybe you could call it `contentToCheck`?

File: nodejs/week4/spam-detector/index.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    //more than 60% uppercase characters
+    let uppercaseCharacters = 0;
+    let uppercaseCharactersArray = [];
+    let numberOfCharacters = subject.length;
+    const subjectWithoutSpaces = subject.split(" ").join();
```
Comment (Hatko) [inline nodejs/week4/spam-detector/index.js]: `subjectWithoutSpaces` contains all the words joined with commas. To substitute all of the spaces by empty strings, you have to pass an empty string to `join`. Also, keep in mind, that after you change it, `numberOfCharacters` will not always be equal `subjectWithoutSpaces.length`, as you will remove spaces. To fix it, you could move declaration `numberOfCharacters` on the line below, and make it equal to `subjectWithoutSpaces.length`

File: nodejs/week4/spam-detector/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const spamWords = ["viagra", "offer", "free", "business proposal"];
+
+class SpamDetector {
+  isSpam(email) {
+    const subject = email.subject;
+    const body = email.body;
+    const string = subject + body;
+
+    //more than 60% uppercase characters
```
Comment (Hatko) [inline nodejs/week4/spam-detector/index.js]: As I understand from the task, you had to check the whole email on uppercase character. Currently you're checking only the subject line.

File: nodejs/week4/spam-detector/index.js (hunk 1 above)
Comment (MirSzabo) [reply nodejs/week4/spam-detector/index.js]: changed

File: nodejs/week4/spam-detector/index.js (hunk 3 above)
Comment (MirSzabo) [reply nodejs/week4/spam-detector/index.js]: True :O 
changed

File: nodejs/week4/spam-detector/index.js (hunk 2 above)
Comment (MirSzabo) [reply nodejs/week4/spam-detector/index.js]: yes makes perfect sense, changed

**Pull request merged**